src/item.cpp

```diff
@@ -480,11 +481,19 @@ bool item::merge_charges( const item &rhs )
     if( !count_by_charges() || !stacks_with( rhs ) ) {
         return false;
     }
+    // assumes that charges are always >= 0 (as negative charges make no sense for items counted by charges)
```

Comment:
Slightly misleading, `charges` is always >= 0 for all items irrespective of whether they are counted by charges or not. Any other value is a bug.